installer/uninstaller.sh

```diff
@@ -33,42 +33,94 @@ done
 
 shift $((OPTIND-1))
 
+
 CHARTS_LIST="$IMM_RELEASE_PREFIX-mgt-api $IMM_RELEASE_PREFIX-crd $IMM_RELEASE_PREFIX-dex $IMM_RELEASE_PREFIX-ingress $IMM_RELEASE_PREFIX-openldap $IMM_RELEASE_PREFIX-minio"
 HELM_LS_OUTPUT=`helm ls --output json`
 HELM_LIST=`jq --arg namearg "Releases" '.[$namearg]' <<< $HELM_LS_OUTPUT`
-K8S_NS_OUTPUT=`kubectl get ns --output=json`
-K8S_NS_LIST=`jq --arg namearg "items" '.[$namearg]' <<< $K8S_NS_OUTPUT`
 helm_arr=()
 K8S_NS_ARR=()
 
+CERTS_PATH="`pwd`/certs/$IMM_RELEASE_PREFIX"
 cd ../scripts
-./imm -k rm t default-tenant
+LIST_TENANTS_RESPONSE=`./imm ls t`
 
-CERTS_PATH="`pwd`/certs/$IMM_RELEASE_PREFIX"
-echo "Certificates marked to delete:"
-tree $CERTS_PATH
-read -r -p "Do you want to delete certificates? [y/N] " response
-if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
-    rm -rf $CERTS_PATH
-fi
-cd ../installer
+if [[ "$LIST_TENANTS_RESPONSE" =~ "Token not valid" ]]; then
+        echo "Uninstallation aborted. Your token is invalid, please log in and try again."
+else
+    while read i; do
+        RELEASE_NAME=`jq --arg namearg "Name" '.[$namearg]' <<< $i | tr -d '"'`
+        mark_release_to_be_deleted $RELEASE_NAME
+    done <<<"$(jq -c '.[]' <<< $HELM_LIST)"
 
-while read i; do
-   echo $i
-   RELEASE_NAME=`jq --arg namearg "Name" '.[$namearg]' <<< $i | tr -d '"'`
-   mark_release_to_be_deleted $RELEASE_NAME
-done <<<"$(jq -c '.[]' <<< $HELM_LIST)"
+    [[ $LIST_TENANTS_RESPONSE =~ '['(.*)']' ]] &&
+        tenants=`echo ${BASH_REMATCH[1]} | tr -d "\""`
+    IFS=', ' read -r -a tenants_arr <<< "$tenants"
 
+    if [[ "$quiet" != "yes" ]]; then
+        echo "Uninstalling IMM. Following components will be removed:"
+        echo "*Helm releases:"
+        for release in "${helm_arr[@]}"; do echo "    - $release" ; done
+        echo "*Tenants with theirs resources(models,endpoints etc.):"
+        for tenant in "${tenants_arr[@]}"; do echo "    - $tenant" ; done
+        echo "Are you sure you want to uninstall IMM? y/N"
+        read DELETE_IMM
+    else
+        DELETE_IMM="y"
+    fi
+    if [[ $DELETE_IMM == "y" ]]; then
+        error=0
+        echo "Deleting tenants..."
+        for tenant in "${tenants_arr[@]}"; do
+            DELETE_TENANT_RESULT=`./imm -k rm t $tenant`
+            if [[ !("$DELETE_TENANT_RESULT" =~ "DELETED") ]]; then
+                error=1
+            fi
+            echo $DELETE_TENANT_RESULT
+        done
 
-echo "Releases marked to be deleted:"
+        if [[ $error == 0 ]]; then
+            echo "Deleting helm releases..."
+            cd ../installer
+            for i in "${helm_arr[@]}"; do
+                HELM_OUTPUT=$((helm del --purge $i) 2>&1)
+                echo $HELM_OUTPUT
+                if [[ $HELM_OUTPUT =~ E|error ]]; then
+                    error=1
+                fi
+            done
+        fi
 
-for i in "${helm_arr[@]}"; do echo "$i" ; done
+        if [[ $error == 0 ]]; then
+            echo "Deleting minio-ingress..."
```

Comment:
In [here](https://github.com/IntelAI/inference-model-manager/blob/da11ed9e312073f268e16b175ae4415e7dc6e348/installer/uninstaller.sh#L37) we have `$IMM_RELEASE_PREFIX-minio`, I wonder what will happen if user has an external minio so uninstaller won't find release with this name, if uninstaller will going to fall with message that not all components was deleted?